File: naive/src/virtual/fib.java
public class fib {

    public static void main(String[] args) {
        
        String number = args[0];
        int n = Integer.parseInt(number);
        int result = fib.done(n);

        System.out.printf("res = %d", result);
    }

    public static int done(int n) {
        if (n < 3) {
            return 1;
        }
        return done(n - 1) + done(n - 2);
    }
}

File: src/loop/Clojure/fib.clj
(defn fib 
  ([n] 
   (fib n (bigint 0) (bigint 1)))
  ([n prev curr]
   (if (< n 3)
     (+ prev curr)
     (recur (- n 1) curr (+ prev curr)))))

(let [
      cmd-arg (first *command-line-args*)
      number (read-string cmd-arg)
      result (str "res = " (fib number))
      ]
  (println result)) 
